src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -595,37 +595,87 @@ internal static int CompareOrdinalIgnoreCase(ref char strA, int lengthA, ref cha
         }
 
 
-        internal static bool EqualsOrdinalIgnoreCase(ref char strA, ref char strB, int length)
+        internal static bool EqualsOrdinalIgnoreCase(ref char charA, ref char charB, int length)
         {
-            ref char charA = ref strA;
-            ref char charB = ref strB;
-
-            // in InvariantMode we support all range and not only the ascii characters.
-            char maxChar = (GlobalizationMode.Invariant ? (char)0xFFFF : (char)0x7F);
-
-            while (length != 0 && charA <= maxChar && charB <= maxChar)
+            if (!GlobalizationMode.Invariant)
             {
-                // Ordinal equals or lowercase equals if the result ends up in the a-z range 
-                if (charA == charB ||
-                    ((charA | 0x20) == (charB | 0x20) &&
-                        (uint)((charA | 0x20) - 'a') <= (uint)('z' - 'a')))
+                IntPtr byteOffset = IntPtr.Zero;
+                while ((uint)length >= 2)
```

Comment:
Can we assume null termination? If so is it safe to read 2 chars even if length is 1, such that you can avoid the `if (length != 0)` block?